### PR TITLE
Improve link labels

### LINKED ISSUE #604: Improve wording for screen readers accessibility
There are some links that refers to `here` or `this website`.  That [doesn't follow good practices](https://webaccess.berkeley.edu/ask-pecan/click-here), they should be more informative.

Examples I've found so far:
- [20-carpentries](https://github.com/carpentries/instructor-training/blob/gh-pages/_episodes/20-carpentries.md]
   - If you would like to host a Software Carpentry workshop at your institution, `visit this website for more information`.
  - Policies related to instructor training and workshops for both Software Carpentry and Data Carpentry can be found `here`.

### PR DESCRIPTION
**Pull request by remram44**
Fixes #604

### CODE REVIEW

File: _extras/checkout.md [hunk 1]
```diff
@@ -218,8 +218,8 @@ they will notify the training coordinator.
 ## Part 3: Teach a Short Demonstration Lesson
 <a name="demo-lesson"></a>
 
-The final step is to sign up for a 5-minute teaching demonstration online using
-[this Etherpad]({{page.demopad}})
+The final step is to sign up for a 5-minute demonstration online using
+[the teaching demo schedule]({{page.demopad}}) Etherpad
```
Comment (ChristinaLK) [inline _extras/checkout.md]: Add "Etherpad" to the link text.

File: _extras/guide.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ re-activate their Trainer status at any time by resuming Trainer activities.
 ## II. Becoming a Trainer
 The Trainers group periodically accepts new members via application. New 
 Trainers undergo an eight-week training program outlined 
-[in this document][trainer-process] and
+[in the trainer process document][trainer-process] and
```
Comment (ChristinaLK) [inline _extras/guide.md]: remove "in" from the link text

**Comment by ChristinaLK** [comment]
cc @karenword can you read over as a second pair of eyes?  If you want to wait until the changes I suggested but it otherwise looks good, let me know and I can merge if the PR is updated.

**Comment by karenword** [comment]
Looks good to me! Thanks @remram44 and thanks also to @dpshelio for the catch!

**Pull request merged**